plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -366,6 +536,69 @@ static int truncate_log(const struct flb_cloudwatch *ctx, const char *log_buffer
     return FLB_FALSE;
 }
 
+/*
+ * Helper function to remove keys prefixed with aws_entity
+ * from a message pack map
+ */
+void remove_key_from_nested_map(msgpack_object_map *nested_map, msgpack_packer *pk, int filtered_fields) {
+    const int remaining_kv_pairs = nested_map->size - filtered_fields;
+    uint32_t j;
+
+    // Pack the updated nested map into the packer, skipping keys in the remove list
+    msgpack_pack_map(pk, remaining_kv_pairs);  // Initial size will adjust in the next loop
+
+    for (j = 0; j < nested_map->size; j++) {
+        msgpack_object_kv nested_kv = nested_map->ptr[j];
+
+        // Check if the current key is in the removal list
+        if (nested_kv.key.type == MSGPACK_OBJECT_STR && nested_kv.key.via.str.size > AWS_ENTITY_PREFIX_LEN && strncmp(nested_kv.key.via.str.ptr, AWS_ENTITY_PREFIX, AWS_ENTITY_PREFIX_LEN) == 0) {
+            // Skip the key in the remove list
+            continue;
+        }
+
+        // Pack the remaining key-value pairs into the packer
+        msgpack_pack_object(pk, nested_kv.key);
+        msgpack_pack_object(pk, nested_kv.val);
+    }
+}
+
+/*
+ * Main function to remove keys prefixed with aws_entity
+ * from the root and nested message pack map
+ */
+void remove_unneeded_field(msgpack_object *root_map, const char *nested_map_key, msgpack_packer *pk,int root_filtered_fields, int filtered_fields) {
+    uint32_t i;
+
+    if (root_map->type == MSGPACK_OBJECT_MAP) {
+        msgpack_object_map root = root_map->via.map;
+
+        // Prepare to pack the modified root map (size may be unchanged or reduced)
+        msgpack_pack_map(pk, root.size-root_filtered_fields);
+
+        for (i = 0; i < root.size; i++) {
+            msgpack_object_kv root_kv = root.ptr[i];
+
+            // Check if this key matches the nested map key (e.g., "kubernetes")
+            if (filtered_fields > 0 &&
+                root_kv.key.type == MSGPACK_OBJECT_STR &&
+                strncmp(root_kv.key.via.str.ptr, nested_map_key, root_kv.key.via.str.size) == 0 &&
+                root_kv.val.type == MSGPACK_OBJECT_MAP) {
+
+                // Pack the nested map key
+                msgpack_pack_object(pk, root_kv.key);
+
+                // Remove the unneeded key from the nested map
+                remove_key_from_nested_map(&root_kv.val.via.map, pk,filtered_fields);
+            } else if (root_filtered_fields > 0 && root_kv.key.type == MSGPACK_OBJECT_STR && root_kv.key.via.str.size > AWS_ENTITY_PREFIX_LEN && strncmp(root_kv.key.via.str.ptr, AWS_ENTITY_PREFIX, AWS_ENTITY_PREFIX_LEN) == 0) {
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions

and line is too long